content/docs/ref-resolving.mdx

```diff
@@ -5,61 +5,55 @@ title: Ref Resolving
-or it might make sense to split up a very big schema into multiple smaller ones for maintenance reasons.
+Resolving of external dependencies is not a core feature of JSON Forms which is why JSON Forms itself can only handle simple `$ref` resolving, which should be sufficient in most cases.
+For a more complex scenario or when resolving references pointing to external schemas, you can use `json-schema-ref-parser` of a similar library to resolve references before passing them to JSON Forms.
 
-If that's the case, you can add a custom ref resolver based on [JSON Schema \$Ref parser](https://github.com/APIDevTools/json-schema-ref-parser)
-that determines how `$ref`s will be resolved. You'll also need to adapt the AJV validator in order to enable validation of the referenced schema.
+Below you can find an example, on how to use `json-schema-ref-parser`:
 
-To do so, use the `refParserOptions` and `ajv` props of the `JsonForms` component and pass in the ref resolver together with the AJV instance.
+```ts
+import React, { useState } from 'react';
+import { JsonForms } from '@jsonforms/react';
+import { materialCells, materialRenderers } from '@jsonforms/material-renderers';
+import $RefParser from '@apidevtools/json-schema-ref-parser';
+import JsonRefs from 'json-refs';
 
-Here's an eample of how to accomplish this. We'll first create a customized AJV instance. `yourSchemaObject` is the actual schema to be referenced
-while `yourSchemaIdentifier` is the id that can be used to reference the schema via a `$ref`.
+import mySchemaWithReferences from 'myschema.json';
 
-```javascript
-const ajv = createAjv();
-ajv.addSchema(yourSchemaObject, 'yourSchemaIdentifier');
-```
-
-Next, we'll set up the ref resolver. The implementation is up to you, but you'll need to implement a `canRead` method that
-determines whether a current ref needs to be resolved and a `read` method that returns the actual schema. The `order` property
-determines the order in which resolvers are executed, until the first one returns true for `canRead`. For more info about ref resolving
-please see the [JSON Schema \$Ref parser documentation](https://apidevtools.org/json-schema-ref-parser/docs/plugins/resolvers.html).
-Note that the name for the actual resolver, which we called `foo` here, does not actually matter.
-
-```javascript
-const yourSchemaResolver = {
-  order: 1,
-  canRead: function(file) {
-    return file.url.indexOf('yourSchemaIdentifier') !== -1;
-  },
-  read: function() {
-    return JSON.stringify(yourSchemaObject);
-  },
-};
 const refParserOptions = {
-  resolve: {
-    foo: yourSchemaResolver,
-  },
-};
+  dereference: {
+    circular: false
+  }
+}
+
+function App() {
+  const [data, setData] = useState(initialData);
+  const [resolvedSchema, setSchema] = useState();
+
+  useEffect(() => {
+    $RefParser.dereference(mySchemaWithReferences).then(res => setSchema(res.$schema));
```

Comment:
You're missing the refParserOptions here. Please check whether the format defined above is the correct one for this call

Reply:
Seems to be correct: https://apitools.dev/json-schema-ref-parser/docs/options.html#example